### ISSUE TITLE
dind image: must align mtu setting with the fn-api container's

### ISSUE THREAD

**Issue by jan-g** [fnproject/fn#684]
The problem does not arise when using an embedded /var/run/docker.sock - or when running with dind on a host that can supply 1500-byte MTUs to containers on the 'host' docker.

However, in many k8s environments the host docker uses an overlay network which'll take bytes away from the effective MTU of outer containers; eg, vxlan needs 50 bytes, often leaving a 1450 MTU on the container running dind and fn-api.

In such an arrangement, packets exceeding the smaller MTU may be invisibly dropped as they travel across the dind's docker0 bridge. This mostly surfaces as a failure of functions to be able to reliably talk to external services. (Note, the failure may be intermittent depending on the profile of the resulting TCP communication.)

A robust fix for this is to intercept the startup of the dind dockerd and ensure that `/etc/docker/daemon.json` (currently absent) contains the following setting:

```json
{
  "mtu": 1450
}
```

(or whatever the MTU on the external interface may be). Ideally this should be autosized so the container works in a variety of deployments.

**Comment by jan-g** [comment]
https://github.com/fnproject/fn/pull/685

**Comment by jan-g** [comment]
Testing:

```sh
% docker network create -o com.docker.network.driver.mtu=1450 fl
% docker run  --net fl --rm --name fnserver -v /var/run/docker.sock:/var/run/docker.sock  fnproject/fnserver
```
and in another window:
```sh
% docker exec -ti fnserver docker run --rm busybox ifconfig
Unable to find image 'busybox:latest' locally
latest: Pulling from library/busybox
fb46b04c527d: Pull complete 
Digest: sha256:436bbf48aa1198ebca8eac0ad9a9c80c8929d9242e02608f76ce18334e0cfe6a
Status: Downloaded newer image for busybox:latest
eth0      Link encap:Ethernet  HWaddr 02:42:AC:11:00:02  
          inet addr:172.17.0.2  Bcast:172.17.255.255  Mask:255.255.0.0
          UP BROADCAST RUNNING MULTICAST  MTU:1450  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

```

**Comment by rdallman** [comment]
closed in #685  thanks!
